src/reactors/PriorityOrderReactor.sol

```diff
@@ -16,31 +16,47 @@ contract PriorityOrderReactor is BaseReactor {
     using PriorityFeeLib for PriorityInput;
     using PriorityFeeLib for PriorityOutput[];
 
+    /// @notice thrown when an order's deadline is in the past
     error InvalidDeadline();
+    /// @notice thrown when an order's auctionStartBlock is in the future
     error OrderNotFillable();
+    /// @notice thrown when an order's input and outputs both scale with priority fee
     error InputOutputScaling();
+    /// @notice thrown when an order's cosignature does not match the expected cosigner
+    error InvalidCosignature();
+    /// @notice thrown when an order's cosigner target block is invalid
+    error InvalidCosignerTargetBlock();
+    /// @notice thrown when an order's min priority fee is greater than the priority fee of the transaction
+    error InsufficientPriorityFee();
 
     constructor(IPermit2 _permit2, address _protocolFeeOwner) BaseReactor(_permit2, _protocolFeeOwner) {}
 
     /// @inheritdoc BaseReactor
-    /// @notice a tx's priority fee must be equal to the difference between the tx gas price and the block's base fee
-    ///         this may not be the case on all chains
+    /// @notice tx.gasprice must be greater than or equal to block.basefee
     function _resolve(SignedOrder calldata signedOrder)
         internal
         view
         override
         returns (ResolvedOrder memory resolvedOrder)
     {
-        PriorityOrder memory priorityOrder = abi.decode(signedOrder.order, (PriorityOrder));
-        _validateOrder(priorityOrder);
+        PriorityOrder memory order = abi.decode(signedOrder.order, (PriorityOrder));
+        bytes32 orderHash = order.hash();
+
+        _updateWithCosignerData(order);
+        _validateOrder(orderHash, order);
 
         uint256 priorityFee = tx.gasprice - block.basefee;
+        if (priorityFee < order.minPriorityFeeWei) {
+            revert InsufficientPriorityFee();
```

Comment:
ah I wonder if these should still be allowed, and go through with 0 tax. I read this as "network is congested, so X prio fee is required to even get included. anything above that is MEV I want to tax". And then the filler pays <X, so either congestion went down or this order has very low MEV, so shouldn't really tax it

Reply:
ah yea I agree these should still go through